### ISSUE TITLE
Fix tests on Node.js 20

### ISSUE THREAD

**Issue by XhmikosR** [dependents/node-dependency-tree#169]
Something changed in Node.js 20 and tests fail on main since a few months now.

If someone figures out why, please make a PR and CC me.

**Comment by GiladShoham** [comment]
I have the same problem.
@XhmikosR which version of node it started?

**Comment by GiladShoham** [comment]
Maybe it's related to this change in 20.5.0
[[986b46a567](https://github.com/nodejs/node/commit/986b46a567)] - fs: add a fast-path for readFileSync utf-8 (Yagiz Nizipli) https://github.com/nodejs/node/pull/48658

**Comment by GiladShoham** [comment]
another relevant reference is here - https://github.com/kubernetes-client/javascript/pull/1202

**Comment by XhmikosR** [comment]
@GiladShoham yeah you are probably right. Unsure how to make tests pass again on Node.js 20. If you have any suggestions feel free to make a PR and CC me.

**Comment by XhmikosR** [comment]
I pushed a branch with debug enabled on CI and the issue seems to be with `fs.existsSync`:

* Node.js 18: https://github.com/dependents/node-dependency-tree/actions/runs/8565544868/job/23473747457
* Node.js 20: https://github.com/dependents/node-dependency-tree/actions/runs/8565544868/job/23473747584

Unsure how to proceed...

We are also hitting this in other repos like node-stylus-lookup:

* https://github.com/dependents/node-stylus-lookup/actions/runs/8565474632/job/23473580973
* https://github.com/dependents/node-stylus-lookup/actions/runs/8565474632/job/23473581057
